src/Nethermind/Nethermind.Evm/CodeAnalysis/IL/Word.cs

```diff
@@ -0,0 +1,328 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Int256;
+using Nethermind.Trie;
+using Newtonsoft.Json.Linq;
+using Org.BouncyCastle.Utilities;
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using System.Text;
+using System.Threading.Tasks;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Nethermind.Evm.CodeAnalysis.IL;
+
+[StructLayout(LayoutKind.Explicit, Size = 32)]
+public struct Word
+{
+    public const int Size = 32;
+    public const int FullSize = 256;
+
+    [FieldOffset(0)] public unsafe fixed byte _buffer[Size];
+
+    [FieldOffset(Size - sizeof(byte))]
+    public byte _uByte0;
+
+    [FieldOffset(Size - sizeof(int))]
+    private int _sInt0;
+
+    [FieldOffset(Size - sizeof(int))]
+    private uint _uInt0;
+
+    [FieldOffset(Size - 2 * sizeof(int))]
+    private uint _uInt1;
+
+    [FieldOffset(Size - 1 * sizeof(ulong))]
+    private ulong _ulong0;
+
+    [FieldOffset(Size - 2 * sizeof(ulong))]
+    private ulong _ulong1;
+
+    [FieldOffset(Size - 3 * sizeof(ulong))]
+    private ulong _ulong2;
+
+    [FieldOffset(Size - 4 * sizeof(ulong))]
+    private ulong _ulong3;
+
+    public bool CheckIfEqual(ref Word other) => _ulong0 == other._ulong0 && _ulong1 == other._ulong1 && _ulong2 == other._ulong2 && _ulong3 == other._ulong3;
+
+    public bool IsZero => (_ulong0 | _ulong1 | _ulong2 | _ulong3) == 0;
+    public bool IsOne => (_ulong1 | _ulong2 | _ulong3) == 0 && (_ulong0 == 0x0100000000000000);
+    public bool IsMinusOne => _ulong1 == ulong.MaxValue && _ulong2 == ulong.MaxValue && _ulong3 == ulong.MaxValue && _ulong0 == ulong.MaxValue;
+    public bool IsP255 => (_ulong3 | _ulong1 | _ulong2) == 0 && (_ulong0 == 0x0000000000000001);
+    public bool IsOneOrZero => (_ulong1 | _ulong2 | _ulong3) == 0 && ((_ulong0 << 1) == 0);
+    public bool IsShort => (_ulong1 | _ulong2 | _ulong3) == 0 && ((_ulong0 << 16) == 0);
+
+    public void ToZero()
+    {
+        _ulong0 = 0; _ulong1 = 0;
+        _ulong2 = 0; _ulong3 = 0;
+    }
+    public unsafe ZeroPaddedSpan ZeroPaddedSpan
+    {
+        set
+        {
+            int startIndex = value.PadDirection == PadDirection.Right ? 0 : value.PaddingLength;
+            fixed (byte* src = value.Span, dest = _buffer)
+            {
+                Buffer.MemoryCopy(src, dest + startIndex, 32, value.Span.Length);
+            }
+        }
+    }
+    public unsafe byte[] Array
+    {
+        get
+        {
+            byte[] array = new byte[32];
+            fixed (byte* src = _buffer, dest = array)
```

Comment:
Why pinning here? We could create the spac from the source or, as we're copying 32, use `Vector256` assignment here.